src/overlay/test/OverlayTests.cpp

```diff
@@ -1928,6 +1934,92 @@ TEST_CASE("overlay pull mode", "[overlay][pullmode]")
         REQUIRE(numUnknownDemand(apps[0]) == 1);
     }
 
+    SECTION("do not advertise to peers that know about tx")
+    {
+        auto root = TestAccount::createRoot(*apps[0]);
+        auto tx = root.tx({txtest::createAccount(
+            txtest::getAccount("acc").getPublicKey(), 100)});
+        auto adv = createAdvert(std::vector<std::shared_ptr<StellarMessage>>{
+            std::make_shared<StellarMessage>(tx->toStellarMessage())});
+        auto twoNodesRecvTx = [&]() {
+            // Node0 and Node1 know about tx0 and will advertise it to Node2
+            REQUIRE(apps[0]->getHerder().recvTransaction(tx, true) ==
+                    TransactionQueue::AddResult::ADD_STATUS_PENDING);
+            REQUIRE(apps[1]->getHerder().recvTransaction(tx, true) ==
+                    TransactionQueue::AddResult::ADD_STATUS_PENDING);
+        };
+
+        SECTION("pull mode enabled on all")
+        {
+            twoNodesRecvTx();
+
+            // Give enough time for Node2 to issue a demand and receive tx0
+            testutil::crankFor(clock,
+                               3 * apps[2]->getConfig().FLOOD_DEMAND_PERIOD_MS +
```

Comment:
What do you think about using an arbitrarily large number like 10 seconds? 

If the default values for `FLOOD_ADVERT_PERIOD_MS` or `FLOOD_DEMAND_PERIOD_MS` change, this test may break. (e.g., If `ADVERT_PERIOD = 200 ms` and `DEMAND_PERIOD = 50 ms`, then the node may not advertise.) 

I had the same thought about line 2000.

Reply:
Updated with a fixed number.